### PR TITLE
Fix for issue #185 - BearerStrategy profileName issue.

### PR DESCRIPTION
**Pull request by brandwe**

### CODE REVIEW

File: lib/bearerstrategy.js [hunk 1]
```diff
@@ -293,8 +293,9 @@ Strategy.prototype.authenticate = function authenticateStrategy(req) {
 
       var metadataURL = self._options.identityMetadata;
 
-      if (self._options.policyName) {
-        log.info('B2C: using policy %s', self._options.policyName);
+      if (!self._options.policyName) {
```
Comment (polita) [inline lib/bearerstrategy.js]: Should we really be erroring any time there's no policyName? What if we're not using B2C- there'd be no policy here, right? And why remove the log info on the policy?

Comment (brandwe) [inline lib/bearerstrategy.js]: You are right, but we have a bigger problem. I had two code paths - one for B2C and one for v2. It seems that this has been removed. A lot is now broken if that is the case, because this code now:

* Automatically does changes to the metadata endpoint (removes common)
* Automatically uses policyName when evaluating policies.

We need to have two forks in the waterfalls like I had in the old code base. I'm open to other options as well.

Right now this is effectively a B2C BearerStrategy. 🎱

Comment (brandwe) [inline lib/bearerstrategy.js]: So what I realize I did when I first added B2C was use policyName as the trigger for doing B2C work, so logging that it isn't there isn't correct behavior. Reverted.

File: lib/bearerstrategy.js [hunk 2]
```diff
@@ -294,7 +294,7 @@ Strategy.prototype.authenticate = function authenticateStrategy(req) {
       var metadataURL = self._options.identityMetadata;
 
       if (self._options.policyName) {
-        log.info('B2C: using policy %s', self._options.policyName);
+      log.info('B2C: using policy %s', self._options.policyName);
```
Comment (polita) [inline lib/bearerstrategy.js]: Why does the indentation look off here? Where are we closing the open bracket on the if line above?

Comment (brandwe) [inline lib/bearerstrategy.js]: As I said before, this is the signal to use the B2C flow, so this is if statement is quite large (it sets the stage for B2C, including creating a new caching key for policy). The end of the if statement is line number 312: https://github.com/AzureAD/passport-azure-ad/blob/e7305c2cfc7f6053fec9d8cbc5797791fff5c7f6/lib/bearerstrategy.js#L312

Comment (lovemaths) [inline lib/bearerstrategy.js]: @brandwe  There should be two extra blanks in front of 297.

Comment (polita) [inline lib/bearerstrategy.js]: @brandwe You should just revert the change on line 297, as @lovemaths says.

**Comment by polita** [comment]
@lovemaths Please test this against a B2C tenant before we merge the change. Looks like we might need a fix to #187 before the E2E will work.
